### PR TITLE
DOC: special.softmax: document how to use a temperature parameter

### PR DESCRIPTION
**Pull request by jpmikhail**
#### Reference issue
N/A

#### What does this implement/fix?

This implements an additional parameter `beta` to the `softmax` function. 

#### Additional information

`beta` is an inverse temperature parameter that changes the weights assigned by `softmax`. The default value, 1, does not change the return value of `softmax`. Values greater than 1 correspond with lower temperature and thus place comparatively greater weight on greater values of the input array, `x`. Likewise, values less than 1 correspond with higher temperatures and thus place comparatively greater weight on lesser values of `x`.

### CODE REVIEW

**Comment by j-bowhay** [comment]
Looks like you have accidentally picked up some other commits alongside your changes

**Comment by jpmikhail** [comment]
> Looks like you have accidentally picked up some other commits alongside your changes

Yep my bad, should be cleaned up now.

**Comment by rgommers** [comment]
@jpmikhail do you have a justification for this addition? I quickly checked PyTorch and JAX, and they both don't have any such parameter:
- https://docs.pytorch.org/docs/main/generated/torch.nn.Softmax.html
- https://docs.jax.dev/en/latest/_autosummary/jax.nn.softmax.html

**Comment by peterbell10** [comment]
Why do you think `softmax(x, beta=beta)` is better than `softmax(x * beta)`?

**Comment by jpmikhail** [comment]
> Why do you think `softmax(x, beta=beta)` is better than `softmax(x * beta)`?

Functionally they're identical, of course. I think having the capability documented along with the function itself is cleaner and makes the use case more obvious. Users unfamiliar with the other contexts of softmax may not assume multiplying the input array by a scalar would be a well-established method of adjusting weights, or they may not know how to effectively communicate the meaning if they did.

Potentially, some additional documentation could be added instead of the parameter if people think that approach is better or more consistent with scipy practices.

**Comment by lucascolley** [comment]
> I think having the capability documented along with the function itself is cleaner and makes the use case more obvious. Users unfamiliar with the other contexts of softmax may not assume multiplying the input array by a scalar would be a well-established method of adjusting weights, or they may not know how to effectively communicate the meaning if they did.
> 
> Potentially, some additional documentation could be added instead of the parameter if people think that approach is better or more consistent with scipy practices.

I think that sounds like a better idea. Would you like to rework the PR to add to the Notes section of the docstring, explaining the functionality?
